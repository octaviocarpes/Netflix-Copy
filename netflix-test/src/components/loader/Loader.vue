<template>
  <div class="loader-container" v-if="loading">
    <div class="lds-facebook">
      <div></div>
      <div></div>
      <div></div>
    </div>
  </div>
</template>

<script>
import store from "@/store";
export default {
  name: "Loader",

  computed: {
    loading() {
      return store.state.loader.loading;
    }
  }
};
</script>

<style lang="scss" scoped>
.loader-container {
  position: absolute;
  width: 100%;
  height: 100%;

  background-color: rgba(0, 0, 0, 0.8);
  .lds-facebook {
    display: inline-block;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 64px;
    height: 64px;
  }
  .lds-facebook div {
    display: inline-block;
    position: absolute;
    left: 6px;
    width: 13px;
    background: #fff;
    animation: lds-facebook 1.2s cubic-bezier(0, 0.5, 0.5, 1) infinite;
  }
  .lds-facebook div:nth-child(1) {
    left: 6px;
    animation-delay: -0.24s;
  }
  .lds-facebook div:nth-child(2) {
    left: 26px;
    animation-delay: -0.12s;
  }
  .lds-facebook div:nth-child(3) {
    left: 45px;
    animation-delay: 0;
  }
  @keyframes lds-facebook {
    0% {
      top: 6px;
      height: 51px;
    }
    50%,
    100% {
      top: 19px;
      height: 26px;
    }
  }
}
</style>
