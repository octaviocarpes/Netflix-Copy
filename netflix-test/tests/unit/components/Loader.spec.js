import Vue from "vue";
import Loader from "@/components/loader/Loader.vue";

describe("Loader.spec.js", () => {
  let cmp, vm;

  beforeEach(() => {
    cmp = Vue.extend(Loader);
    vm = new cmp({}).$mount();
  });

  it("has the expected html structure", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});