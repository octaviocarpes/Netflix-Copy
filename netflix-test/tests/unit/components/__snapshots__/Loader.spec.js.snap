// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader.spec.js has the expected html structure 1`] = `<!---->`;
